### PR TITLE
Move WorkstationConfigLoader into chef-config

### PR DESCRIPTION
**Pull request by mcquin**
This is needed for [ohai config rfc](https://github.com/chef/chef-rfc/blob/master/rfc053-ohai-config.md) so that Ohai can find local workstation config files.

@chef/client-maintainers @chef/client-core

### CODE REVIEW

File: chef-config/lib/chef-config/exceptions.rb [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 module ChefConfig
 
-  class InvalidPath < StandardError
-  end
+  class ConfigurationError < ArgumentError; end
```
Comment (coderanger) [inline chef-config/lib/chef-config/exceptions.rb]: Do we want put subclass stubs in Chef that inherit from these for API compat?

Comment (mcquin) [inline chef-config/lib/chef-config/exceptions.rb]: @danielsdeleo @jkeiser ?

Comment (mcquin) [inline chef-config/lib/chef-config/exceptions.rb]: @coderanger [this should take care of it](https://github.com/chef/chef/blob/6a4f414666b3688a3dfa94b131f83235f4d2fa64/lib/chef/exceptions.rb#L28)

Comment (coderanger) [inline chef-config/lib/chef-config/exceptions.rb]: :+1:

**Comment by danielsdeleo** [comment]
:+1: awesome :100:

**Comment by tyler-ball** [comment]
:+1:

**Comment by mcquin** [comment]
I believe this requires a release of the chef-config gem for Ohai to be able to use these changes.

**Pull request merged**